src/main/java/com/softserve/bookstore/repositories/UserRepository.java

```diff
@@ -2,123 +2,148 @@
                 .filter(user -> email.equals(user.getEmail().trim()))
                 .findFirst();
     }
 
-    public void add(List<User> users) throws SQLException {
+    public void addUsers(List<User> users) throws SQLException{
         Connection connection = connectionManager.getConnection();
         PreparedStatement userStatement = connection.prepareStatement(INSERT_USERS);
         users.forEach(user -> {
-            UserUtility.addUserToBatch(userStatement, user);
+            addUserToBatch(userStatement, user);
         });
         userStatement.executeBatch();
 
         List<User> lastUsersAdded = findLastUsersAdded(users.size());
         lastUsersAdded.forEach(user -> {
             try {
-                roleRepository.addRole(Role.USER.toString(), user);
-                getUserByEmail(users, user.getEmail())
-                        .orElseThrow(() -> new UserNotFoundException("User does not exist."));
-
-                User currentUser = getUserByEmail(users, user.getEmail()).get();
-
-                if (currentUser.getRoles().size() > 1) {
-                    roleRepository.addRole(Role.ADMIN.toString(), user);
-                }
-                if (currentUser.getOrders().size() > 1) {
-                    addOrdersToUser(currentUser, users.size());
+                    addRole(Role.USER.toString(), user);
+                    getUserByEmail(users, user.getEmail())
+                            .orElseThrow(() -> new UserNotFoundException("User with email"));
+
+                    User currentUser = getUserByEmail(users, user.getEmail()).get();
+
+                    if (currentUser.getRoles().size() > 1) {
+                        addRole(Role.ADMIN.toString(), user);
+                    }
+                    if (currentUser.getOrders().size() > 1) {
+                        addOrdersToUser(currentUser, users.size());
+                    }
+                } catch (UserNotFoundException | SQLException e) {
+                    throw new RuntimeException(e);
                 }
-            } catch (UserNotFoundException | SQLException e) {
-                throw new RuntimeException(e);
-            }
-        });
+            });
     }
 
 
-    public List<User> findLastUsersAdded(int numberOfRecords) throws SQLException {
-        Connection connection = connectionManager.getConnection();
-        PreparedStatement userStatement = connection.prepareStatement(SELECT_LAST_USERS);
-        userStatement.setInt(1, numberOfRecords);
-        ResultSet resultSet = userStatement.executeQuery();
-        List<User> users = UserUtility.getUsersFromResultSet(resultSet);
-        Collections.reverse(users);
+    private  List<User> getUsersFromResultSet(ResultSet resultSet) throws SQLException {
+        List<User> users = new ArrayList<>();
+        while(resultSet.next()){
+            int userId = resultSet.getInt(User.FIELD_USER_ID);
+            String email = resultSet.getString(User.FIELD_EMAIL).trim();
```

Comment:
I used trim because I had problems when taking the values from the ResultSet. In the case of Strings, there is a lot of space after them. I think the problem arises because for the email column, for example, I have the type varchar(50) and an email has 10 characters, from the resultSet it takes my email followed by a space of 40 characters.  You can see it in the screenshot down below
![Capture](https://user-images.githubusercontent.com/49956918/228266500-585f81cc-24e3-4b69-8fb5-5e5a59f1a49b.PNG)